### PR TITLE
NewManagedIdentityCredential returns an error for unsupported ID options

### LINKED ISSUE #23179: Cloud Shell ManagedIdentitySource should throw if a user-assigned managed identity clientID or resourceID is supplied
If the Cloud Shell ManagedIdentitySource is detected via MSAL and a user-assigned managed identity clientID or resourceID is supplied, the ManagedIdentityCredential should throw.

The rationale for taking this minor breaking change is that CloudShell does not support specifying a clientID or ResourceID and the current behavior of silently falling back to attempting to use a system-assigned identity could be unexpected.

### PR DESCRIPTION
**Pull request by chlowell**
Closes #23179

### CODE REVIEW

File: sdk/azidentity/managed_identity_credential_test.go [hunk 1]
```diff
@@ -673,3 +673,31 @@ func TestManagedIdentityCredential_ServiceFabric(t *testing.T) {
 	}
 	testGetTokenSuccess(t, cred)
 }
+
+func TestManagedIdentityCredential_UnsupportedID(t *testing.T) {
```
Comment (jhendrixMSFT) [inline sdk/azidentity/managed_identity_credential_test.go:677]: Does Cloud Shell need a test in here?

Comment (chlowell) [reply sdk/azidentity/managed_identity_credential_test.go:677]: Yes, thank you! I also added one for the affected DefaultAzureCredential scenario.

**Review by jhendrixMSFT — APPROVED** [review]

File: sdk/azidentity/managed_identity_credential.go [hunk 2]
```diff
@@ -60,9 +71,10 @@ func (r ResourceID) String() string {
 type ManagedIdentityCredentialOptions struct {
 	azcore.ClientOptions
 
-	// ID is the ID of a managed identity the credential should authenticate. Set this field to use a specific identity
-	// instead of the hosting environment's default. The value may be the identity's client ID or resource ID, but note that
-	// some platforms don't accept resource IDs.
+	// ID of a managed identity the credential should authenticate. Set this field to use a specific identity instead of
+	// the hosting environment's default. The value may be the identity's client ID or resource ID.
+	// NewManagedIdentityCredential returns an error when the hosting environment doesn't support user-assigned managed
+	// identities, or the specified kind of ID.
```
Comment (RickWinter) [inline sdk/azidentity/managed_identity_credential.go:77]: Should we have miObjectID?

Comment (chlowell) [reply sdk/azidentity/managed_identity_credential.go:77]: Not in this version 😁

**Pull request merged**